src/test/ui/suggestions/suggest-move-types.stderr

```diff
@@ -1,81 +1,93 @@
+   |                                    ^----^^-^^--^
+   |                                     |     |  |
+   |                                     |     |  this generic argument must come before the first constraint
+   |                                     |     this generic argument must come before the first constraint
+   |                                     the first constraint is provided here
 
-error: associated type bindings must be declared after generic parameters
-  --> $DIR/suggest-move-types.rs:41:28
+error: constraints in a path segment must come after generic arguments
+  --> $DIR/suggest-move-types.rs:40:27
    |
 LL | struct B<T, U, V, M: Three<A=(), B=(), C=(), T, U, V>> {
-   |                            ----^^----^^----^^^^^^^^^
-   |                            |     |     |
-   |                            |     |     this associated type binding should be moved after the generic parameters
-   |                            |     this associated type binding should be moved after the generic parameters
-   |                            this associated type binding should be moved after the generic parameters
+   |                           ^----^^^^^^^^^^^^^^-^^-^^-^
+   |                            |                 |  |  |
+   |                            |                 |  |  this generic argument must come before the first constraint
+   |                            |                 |  this generic argument must come before the first constraint
+   |                            |                 this generic argument must come before the first constraint
+   |                            the first constraint is provided here
 
-error: associated type bindings must be declared after generic parameters
-  --> $DIR/suggest-move-types.rs:48:53
+error: constraints in a path segment must come after generic arguments
+  --> $DIR/suggest-move-types.rs:48:52
    |
 LL | struct Bl<'a, 'b, 'c, T, U, V, M: ThreeWithLifetime<A=(), B=(), C=(), T, U, V, 'a, 'b, 'c>> {
-   |                                                     ----^^----^^----^^^^^^^^^^^^^^^^^^^^^
-   |                                                     |     |     |
-   |                                                     |     |     this associated type binding should be moved after the generic parameters
-   |                                                     |     this associated type binding should be moved after the generic parameters
-   |                                                     this associated type binding should be moved after the generic parameters
+   |                                                    ^----^^^^^^^^^^^^^^-^^-^^-^^--^^--^^--^
+   |                                                     |                 |  |  |  |   |   |
+   |                                                     |                 |  |  |  |   |   this generic argument must come before the first constraint
```

Comment:
Until I got to this point, I didn't mind the multiple identical messages so much. But now that I see this example, I think we'd rather say `these generic arguments must come before the first constraint` when there is more than one, and use the spans of all of them.

Reply:
Made some tweaks in this direction.